### ISSUE TITLE
Large userscripts and `@require` tags delay page loading

### ISSUE THREAD

**Issue by j4k0xb** [Tampermonkey/tampermonkey#1486]
## Expected Behavior

`@require` of a large file should be roughly as fast as fetch+eval (about 1s)

## Actual Behavior

`@require` makes the entire browser unresponsive for several minutes. (page doesn't load, can't reload other tabs, clicking on the tampermonkey icon doesn't open it, etc.)

![image](https://user-images.githubusercontent.com/55899582/163683198-55bd66f3-7422-4ea0-8ae1-35e1702b5c0b.png)

## Specifications

- Chromium: 100.0.4896.127 (Brave 1.37.116)
- TM: 4.16
- OS: Manjaro Linux 21.2.6

## Script

```js
// ==UserScript==
// @name         Fetch test
// @version      0.1
// @description  try to take over the world!
// @match        https://example.com/
// @grant        none
// ==/UserScript==

fetch('http://localhost:3000/out.js')
  .then(res => res.text())
  .then(eval);
```

```js
// ==UserScript==
// @name         Require test
// @version      0.1
// @description  try to take over the world!
// @match        https://example.com/
// @grant        none
// @require      http://localhost:3000/out.js
// ==/UserScript==
```

`out.js` is 1.5Mb in size, I also uploaded it [here](https://gist.githubusercontent.com/j4k0xb/3a5f95f12e97d26ad5ba67464303b6d7/raw/dbc470c7a807be60fe2d5eb63469e76e5fb34e57/out.js) and it was generated like this:

- app.jsx:
```jsx
import * as React from 'react';
import * as Server from 'react-dom/server';

let Greet = () => <h1>Hello, world!</h1>;
console.log(Server.renderToString(<Greet />));
```
- package.json:
```json
{
  "scripts": {
    "build": "esbuild app.jsx --bundle --sourcemap=inline --outfile=out.js",
  },
  "dependencies": {
    "esbuild": "^0.14.36",
    "react": "^18.0.0",
    "react-dom": "^18.0.0"
  }
}
```

**Comment by scholtzm** [comment]
I can confirm this. Started happening after last TM update.

**Comment by derjanb** [comment]
> @require makes the entire browser unresponsive for several minutes. (page doesn't load, can't reload other tabs, clicking on the tampermonkey icon doesn't open it, etc.)

Looks like Tampermonkey is busy doing something and that's why webRequest hooks can't be processed and pages are not loading anymore. 

Can you please check [Tampermonkey's background console](https://www.tampermonkey.net/faq.php#Q600) for errors or a hint, what's going on?

Your test script including `out.js` `@require` is working fine here.

> I'm wondering if it's related to this change in TM v4.16:

No, `file://` `@require`s have never been cached and other `@require`s were updated afterwards only.

**Comment by j4k0xb** [comment]
Tested again with a smaller (260KB) file which takes 35s

Here is the performance measure:
![image](https://user-images.githubusercontent.com/55899582/164084823-a963b122-7918-4b4a-917d-7cb9a53f8188.png)
![image](https://user-images.githubusercontent.com/55899582/164084878-5c6122c1-f823-4c7b-9cdc-9ecc2ff4bdc6.png)

And the debug console logs:
![image](https://user-images.githubusercontent.com/55899582/164085451-0672cae7-2120-4e9c-97ca-6bf4aff7735e.png)

**Comment by scholtzm** [comment]
Pretty much the same on my end. I just looked at the log and in my case it took a whopping 350 seconds to load the page.

`[Violation] 'load' handler took 350445ms`

**Comment by derjanb** [comment]
I found the root cause. As a workaround you can set "GM/FF compatibility options > Add GM functions to this or window" at the script settings to "off" (or "on" in case it uses "window.GM_someFunction" or "this.GM.someFunction".

![image](https://user-images.githubusercontent.com/767504/164332725-f4821480-a312-493f-851e-cc27ea99e044.png)

**Comment by derjanb** [comment]
Should be fixed at 4.17.6161 (in review|[crx](https://tampermonkey.net/crx/tampermonkey_beta_current.crx)|[xpi](https://firefox.tampermonkey.net/firefox-current-beta.xpi))

Chrome/Edge users, please download the crx file linked above and drag and drop it to the extensions page `chrome://extensions`.
Firefox users please install the [BETA version](https://www.tampermonkey.net/index.php?browser=firefox) or check for BETA version updates at `about:addons` 

For a quick fix please export your settings and scripts as zip or (JSON) file at the "Utilities" tab and import it back at the fixed BETA version.
